### PR TITLE
docs: update external command documentation

### PR DESCRIPTION
**Pull request by Rylan12**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?
- [x] Have you successfully run `brew man` locally and committed any changes?

-----

Follow up to https://github.com/Homebrew/brew/pull/10288

This PR updates the documentation around external commands and how to use the new `named_args` method.

My ultimate goal is to have completions be automatically generated based on `named_args` and to have an e.g. `brew generate-completions` command that will generate completions automatically for a tap. Given this, let me know if it's better for this PR to wait until that's ready (in order to document both at the same time).

### CODE REVIEW

File: docs/External-Commands.md
Comment (MikeMcQuaid) [inline docs/External-Commands.md:88]: Not blocking this PR but: this is the part I'm wondering if we could just generate based on the `named_args` work you've done? In a bunch of cases it ends up being incorrect/outdated.

**Review by MikeMcQuaid — APPROVED** [review]
Great work again 👏🏻

File: docs/External-Commands.md
Comment (EricFromCanada) [inline docs/External-Commands.md:130]:
```suggestion
Use the `named_args` method to specify the type and number of named arguments that are expected. Pass either a symbol to indicate the type of argument expected, an array of symbols to indicate that multiple types should be expected, or an array of strings to specify which specific options should be expected (see the [`brew analytics`](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/cmd/analytics.rb) command for an example of this).
```

File: docs/External-Commands.md
Comment (Rylan12) [reply docs/External-Commands.md:88]: Yeah, I'd like to be able to do that. It is on my list of things to do with the `named_args` stuff. I'm prioritizing the completions, though, so I was planning on doing this later. Let me know if I should reprioritize.

**Comment by Rylan12** [comment]
Good suggestions, thanks, @EricFromCanada!

File: docs/External-Commands.md
Comment (Rylan12) [reply docs/External-Commands.md:88]: Actually, this is probably much simpler than I thought. I definitely want to get the bash completions done soon but maybe I'll take a quick break for this... I'll think about it

**Comment by Rylan12** [comment]
Putting this on hold for now until #10342 is merged so that those changes can be reflected here.

**Comment by Rylan12** [comment]
Okay, now that #10342 has been merged, I've updated this PR to include information about auto-generated usage strings. I'll leave this open for now for any last feedback and I'll plan on merging sometime tomorrow.

**Comment by MikeMcQuaid** [comment]
LGTM 👍🏻

**Pull request merged**
